dali/pipeline/operator/builtin/external_source_test.cc

```diff
@@ -98,34 +104,83 @@ class ExternalSourceTest : public::testing::WithParamInterface<int>,
     return dynamic_cast<ExternalSource<GPUBackend> *>(this->graph_.Node(OpType::GPU, 0).op.get());
   }
 
+  TensorShape<> GetRandShape(int dims) {
+    vector<Index> shape(dims, 0);
+    for (auto &val : shape) {
+      val = this->RandInt(1, 15);
+    }
+    return shape;
+  }
+
+  template<typename T>
+  void FeedWithCpuVector(T *src_op) {
```

Comment:
Isn't that template overly generic?
```suggestion
  template<typename Backend>
  void FeedWithCpuVector(ExternalSource<Backend> *src_op) {
```

Reply:
Done